floor/floor/util/simple_profile.py

```diff
@@ -0,0 +1,86 @@
+import time
+import os
+import logging
+
+
+class profile(object):
+    """Utility decorator that implements a simple profiler.
+    """
+
+    PROFILE_ENABLED = 'PROFILE' in os.environ and os.environ['PROFILE']
+    PROFILE_DATA = []
```

Comment:
So since the same `profile` is applied/used on all profiled methods its functionally equivalent to a singleton right?  Plus that one `profile` still needs to keep track of all the timing data for all functions.

Making it a singleton actually made things a lot cleaner I think.  I made an `instance` method and pulled up a bunch of functionality into the `profile` class which has made it a lot more tidy.

This profiler is barely legitimate as it makes a lot of assumptions on how its going to be used, so I'm inclined to keep it as is until we feel like there's another good use case that can't be solved with the real profiler.  My main goal is for this thing to get as out of the way as possible when its not enabled.